### PR TITLE
Migrate Travis to GitHub actions

### PR DESCRIPTION
**Pull request by airvzxf**
Regarding the issue: https://github.com/scanmem/scanmem/issues/421. I created one action to set up the Linux VM. Furthermore, I created two workflows, one for build and test the application, the other for run the coverity scan.

As my workflow actions showed in this link https://github.com/airvzxf/scanmem/actions, I ran successfully the build, test and coverity scan. This last one was submitted to the build in the Synopsys https://scan.coverity.com/projects/airvzxf-scanmem?tab=overview.

New issues recommended creating:

- Another issue that we can create is related to create the CI/CD for Android, building in Ubuntu Linux.

- Crate issue related to the release versions. We need to discuss the release versions, it is possible to set up the action to generate the release and the artifacts for the release (source code in zip and tar.gz). This set up, could be generated when a tag is adding to some specific commit. I recommend to work in it in a separate issue and pull request, to keep the work simple and focus.

### CODE REVIEW

**Comment by sriemer** [comment]
@airvzxf Thanks for your work on this. :smiley: Can you please squash commits and use the usual imperative (as if you would give commands to the code base to change its behavior) to write git commit descriptions? TIA
~"Added"~ -> "Add", ...

**Comment by 12345ieee** [comment]
Coverity scan should not run on arbitrary push and especially not on PRs.

It should either be manually triggeed, or triggered by a push on a certain branch (not main).

**Comment by airvzxf** [comment]
> Coverity scan should not run on arbitrary push and especially not on PRs.
> 
> It should either be manually triggeed, or triggered by a push on a certain branch (not main).

I'll change for manual run.

Question, what branch name or names, will be trigger the coverity scan?

**Comment by airvzxf** [comment]
> Coverity scan should not run on arbitrary push and especially not on PRs.
> 
> It should either be manually triggeed, or triggered by a push on a certain branch (not main).

I added the change in commit [c982bd6](https://github.com/scanmem/scanmem/pull/422/commits/c982bd6ef409da749e24c1a0b876142f5405b14e).

It runs in two scenarios: manually and if the branch name contains the name coverity (`branches: '**coverity**'`). For more information about the single asterisk or double, here is the reference: [Filter pattern cheat sheet - Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)

I attached the evidence showing the execution of coverity scan when the branch contains the word 'coverity'. Otherwise, it didn't execute the coverity scan:

github-action-coverity-scan

**Comment by airvzxf** [comment]
> @airvzxf Thanks for your work on this. smiley Can you please squash commits and use the usual imperative (as if you would give commands to the code base to change its behavior) to write git commit descriptions? TIA ~"Added"~ -> "Add", ...

@sriemer,

I can squash commits manually, but it is better to do in the pull request because it is easy for the maintainer to add the last commit and squash them. Avoiding the manual process that I need to do, which is reset the commits and commit again after every commit in my local branch. [This solution on StackOverflow showing the steps to squash commits manually](https://stackoverflow.com/questions/5189560/how-do-i-squash-my-last-n-commits-together/61171280#61171280).

---

[GitHub Docs: Configuring commit squashing for pull requests](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests).

GitHub-squash-commits-from-pull-request

---

I don't have permissions to do the squash commits in the pull request, as I showed in the screenshot. Only who has permissions to merge can perform the squash commits.

Mereging-is-blocked

File: .github/workflows/01-build.yml
Comment (12345ieee) [inline .github/workflows/01-build.yml:18]: Nice touch, I expect it to never break, but may as well get this simple install test done.

**Review by 12345ieee — APPROVED** [review]

**Comment by sriemer** [comment]
Looks good. @12345ieee has more experience with this than me. So I'll let him merge this. Thanks for the contribution. :smile:

**Pull request closed without merging**
